### PR TITLE
MAT-2267 Fix get_step_index

### PR DESCRIPTION
**Pull request by patrickherring-TRI**
This PR addresses an assertion error in the get_step_index function used featurization.
- Change the condition for the 'hppc_final_discharge' step

### CODE REVIEW

File: beep/features/featurizer_helpers.py [hunk 1]
```diff
@@ -370,11 +370,16 @@ def get_hppc_ocv(processed_cycler_run, diag_pos):
     cycle_hppc_0 = cycle_hppc.loc[cycle_hppc.cycle_index == cycles[0]]
     #     in case that cycle 2 correspond to two cycles one is real cycle 2, one is at the end
     cycle_hppc_0 = cycle_hppc_0.loc[cycle_hppc_0.test_time < 250000]
-
-    step = 11
-    step_later = 43
-
-    voltage_1 = get_hppc_ocv_helper(cycle_hppc_0, step)
+    first_diagnostic_steps = get_step_index(processed_cycler_run,
```
Comment (moorepatrick) [inline beep/features/featurizer_helpers.py:373]: These calls seem to be duplicates. Do we need all of them or could we do 
```
diagnostic_steps = get_step_index(processed_cycler_run, cycle_type="hppc", diag_pos=0)
step_first = diagnostic_steps['hppc_long_rest']
step_later = diagnostic_steps['hppc_long_rest']
```

**Review by moorepatrick — CHANGES_REQUESTED** [review]
One question about code duplication otherwise LGTM.

File: beep/features/featurizer_helpers.py [hunk 2]
```diff
@@ -415,13 +420,13 @@ def get_chosen_df(processed_cycler_run, diag_pos):
     return chosen
 
 
-def res_calc(chosen, diag_pos, soc, step_ocv, step_cur, index):
+def res_calc(chosen, steps, soc, step_ocv, step_cur, index):
```
Comment (chirranjeevigopal-TRI) [inline beep/features/featurizer_helpers.py:423]: While you are on this, can we rename `chosen` to something more descriptive, say `df_diag` or such?

File: beep/features/featurizer_helpers.py [hunk 3]
```diff
@@ -370,11 +370,16 @@ def get_hppc_ocv(processed_cycler_run, diag_pos):
...
-    step = 11
-    step_later = 43
-
-    voltage_1 = get_hppc_ocv_helper(cycle_hppc_0, step)
+    first_diagnostic_steps = get_step_index(processed_cycler_run,
+                                            cycle_type="hppc",
+                                            diag_pos=0)
+    later_diagnostic_steps = get_step_index(processed_cycler_run,
+                                            cycle_type="hppc",
+                                            diag_pos=0)
```
Comment (chirranjeevigopal-TRI) [inline beep/features/featurizer_helpers.py:378]: Should `diag_pos` be the method arg `diag_pos` for `later_diagnostic_steps` ?

File: beep/features/featurizer_helpers.py (hunk 2 above)
Comment (patrickherring-TRI) [reply beep/features/featurizer_helpers.py]: done

**Review by chirranjeevigopal-TRI — APPROVED** [review]
Looks good to me! Thank you for the changes.

File: beep/features/featurizer_helpers.py (hunk 3 above)
Comment (patrickherring-TRI) [reply beep/features/featurizer_helpers.py]: yes, I think that's the correct usage here

File: beep/features/featurizer_helpers.py [hunk 4]
```diff
@@ -370,13 +370,18 @@ def get_hppc_ocv(processed_cycler_run, diag_pos):
...
     #     in case that cycle 2 correspond to two cycles one is real cycle 2, one is at the end
     cycle_hppc_0 = cycle_hppc_0.loc[cycle_hppc_0.test_time < 250000]
-
-    step = 11
-    step_later = 43
-
-    voltage_1 = get_hppc_ocv_helper(cycle_hppc_0, step)
-    chosen = cycle_hppc.loc[cycle_hppc.cycle_index == cycles[diag_pos]]
-    voltage_2 = get_hppc_ocv_helper(chosen, step_later)
+    first_diagnostic_steps = get_step_index(processed_cycler_run,
```
Comment (patrickherring-TRI) [inline beep/features/featurizer_helpers.py:373]: @moorepatrick Fixing code duplication issue

**Review by moorepatrick — APPROVED** [review]

**Pull request merged**